File: frontend/src/microservices/content/objects/GetReviewSummaryPlannerRequest.ts
// GetReviewSummaryPlannerRequest：content 域资源评论摘要请求对象。

export type GetReviewSummaryPlannerRequest = {
  userId: string
  resourceType: string
  resourceId: string
}

export const getReviewSummaryPlannerRequestFromJson = (json: string): GetReviewSummaryPlannerRequest =>
  JSON.parse(json) as GetReviewSummaryPlannerRequest

export const getReviewSummaryPlannerRequestToJson = (value: GetReviewSummaryPlannerRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/content/objects/GetReviewSummaryPlannerRequest.scala
package com.typesafe.travel.content.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class GetReviewSummaryPlannerRequest(
    userId: String,
    resourceType: String,
    resourceId: String
)
object GetReviewSummaryPlannerRequest:
  given sourceEncoder: Encoder[GetReviewSummaryPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[GetReviewSummaryPlannerRequest] = deriveDecoder
